scipy/stats/_continuous_distns.py

```diff
@@ -3741,7 +3741,15 @@ def _mode(self, p, b):
             return (np.sqrt((1 - p)**2 + b**2) - (1 - p)) / b
 
     def _munp(self, n, p, b):
-        return sc.kve(p + n, b) / sc.kve(p, b)
+        m = sc.kve(p + n, b) / sc.kve(p, b)
```

Comment:
For this to work the way you expect, you'll have to check the numerator and denominator separately.  With the current implementation, with a call such as`geninvgauss.mean(230, 4)`, *both* terms are infinite, so the warning `RuntimeWarning: invalid value encountered in true_divide` is generated by the division and the result is `nan`.  The end result is the same, but instead of the informative warning that you have below, we get a generic warning that doesn't tell the user anything useful.

Reply:
True, I changed it now